integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/HiveTableRelationVisitor.java

```diff
@@ -35,8 +35,13 @@ public HiveTableRelationVisitor(@NonNull OpenLineageContext context, DatasetFact
 
   @Override
   public List<D> apply(LogicalPlan x) {
+    if (!context.getSparkSession().isPresent()) {
```

Comment:
Would it make sense to make `PathUtils.fromCatalogTable` get `OpenLineageContext` as parameter and return `Optional<DatasetIdentifier>`? This would allow having a single session existence check.

Reply:
IMHO there is no difference between having method like this:
```java
DatasetIdentifier fromCatalogTable(CatalogTable, SparkSession) {
  ...
}

////

if (!table.isPresent() || !sparkSession.isPresent()) {
  return Collections.emptyList();
}

di = fromCatalogTable(table.get(), sparkSession.get());
return Collections.singletonList(outputDataset().getDataset(di))
```
and like this:
```java
Optional<DatasetIdentifier> fromCatalogTable(Optional<CatalogTable>, Optional<SparkSession>) {
   ...
}

////

di = fromCatalogTable(table, sparkSession);
if (di.isPresent()) {
  return Collections.singletonList(outputDataset().getDataset(di.get()));
} else {
  return Collections.emptyList();
}
```

Passing the entire OpenLineageContext instead of SparkSession is just the same as last method.